### PR TITLE
Update install guides for 1.4.0

### PR DESCRIPTION
**Pull request by tgraf**
This brings all install guides up to 1.4.0 including:

 * Dedicated guides for etcd-operator and external etcd
 * New EKS guide with eksctl and CoreDNS section based on new cilium-etcd-operator
 * New GKE guide using gcloud, auto-generated YAML and new DaemonSet for node initialization that works with auto-scaling
 * Re-restructure all getting started guides into sections
 * Consolidate all installation guides in one place

Related: https://github.com/cilium/cilium/projects/25

Fixes: #6783

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6812)

### CODE REVIEW

**Comment by tgraf** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-me-please

File: examples/kubernetes/node-init/node-init.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  template:
+    metadata:
+      labels:
+        app: cilium-node-init
+    spec:
+      hostPID: true
+      containers:
+        - name: node-init
+          image: gcr.io/google-containers/startup-script:v1
+          imagePullPolicy: Always
```
Comment (aanm) [inline examples/kubernetes/node-init/node-init.yaml]: imagePullPolicy: IfNotPresent

File: Documentation/kubernetes/requirements.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+Kubernetes Version
+==================
+
+The following Kubernetes versions have been tested in the continuous integration
+system for this version of Cilium:
+
++---------------------+---------------------------------------------------------------+
+| Option              | Notes                                                         |
++---------------------+---------------------------------------------------------------+
+| 1.9                 |                                                               |
```
Comment (aanm) [inline Documentation/kubernetes/requirements.rst]: Missing 1.8

File: Documentation/gettingstarted/k8s-install-etcd-operator.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+.. _k8s_install_etcd_operator:
+
+*********************
+Standard Installation
+*********************
+
+This guides takes you through the steps required to set up Cilium on Kubernetes
+using the cilium-etcd-operator. The cilium-etcd-operator replaces the
+requirement for an external kvstore, you can learn more about it in the section
```
Comment (ianvernon) [inline Documentation/gettingstarted/k8s-install-etcd-operator.rst]: kvstore. You can learn more about it...

File: Documentation/gettingstarted/k8s-install-etcd-operator.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+:ref:`k8s_what_is_the_cilium_etcd_operator`. It is suitable for small and
+medium scale deployments where etcd performance is not absolutely essential.
+Please refer to the  :ref:`k8s_etcd_operator_limitations` section below to
+learn about the exact limitations of this deployment method.
+
+Should you encounter any issues during the installation, please refer to the
+:ref:`troubleshooting_k8s` section and / or seek help on `Slack channel`.
```
Comment (ianvernon) [inline Documentation/gettingstarted/k8s-install-etcd-operator.rst]: on the \`Slack channel\`

File: Documentation/gettingstarted/k8s-install-external-etcd.rst [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+.. _admin_install_daemonset:
+.. _k8s_install_standard:
+
+*******************************
+Installation with external etcd
+*******************************
+
+This guide walks you through the steps required to set up Cilium on Kubernetes
+using an external etcd. Use of an external etcd provides better performance and
+is suitable for larger environment. If you are looking for a simple
```
Comment (ianvernon) [inline Documentation/gettingstarted/k8s-install-external-etcd.rst]: environments

File: Documentation/gettingstarted/k8s-install-kops.rst [hunk 6]
```diff
@@ -62,23 +69,32 @@ Assuming you have all the prerequisites, run the following commands to create th
         aws iam create-access-key --user-name kops
 
 
-``kops`` requires the creation of a dedicated s3 bucket in order to store the state and representation of the cluster. You will need to change the bucket name and provide your unique bucket name (for example a reverse of FQDN added with short description of the cluster). Also make sure to use the region where you will be deploying the cluster.
+``kops`` requires the creation of a dedicated s3 bucket in order to store the
```
Comment (ianvernon) [inline Documentation/gettingstarted/k8s-install-kops.rst]: s3 --> S3

File: Documentation/gettingstarted/k8s-install-kops.rst [hunk 7]
```diff
@@ -94,8 +110,12 @@ In this guide, we will use etcd version 3.1.11 and the latest CoreOS stable imag
 Creating a Cluster
 ====================
 
-* Note that you will need to specify the ``--master-zones`` and ``--zones`` for creating the master and worker nodes. The number of master zones should be odd (1, 3, ...) for HA. For simplicity, you can just use 1 region.
-* The cluster ``NAME`` variable should end with ``k8s.local`` to use the gossip protocol. If creating multiple clusters using the same kops user, then make cluster name unique by adding a prefix such as ``com-company-emailid-``.
+* Note that you will need to specify the ``--master-zones`` and ``--zones`` for
+  creating the master and worker nodes. The number of master zones should be
+  * odd (1, 3, ...) for HA. For simplicity, you can just use 1 region.
+* The cluster ``NAME`` variable should end with ``k8s.local`` to use the gossip
+  protocol. If creating multiple clusters using the same kops user, then make
```
Comment (ianvernon) [inline Documentation/gettingstarted/k8s-install-kops.rst:117]: then make the

File: Documentation/gettingstarted/k8s-installers.rst [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+**********************
+Installer Integrations
+**********************
+
+The following list includes the Kubernetes installer integrations we are aware
+of. If your preferred installer is not in the list, you can always fall back to
+the standard :ref:`k8s_install_etcd_operator` guide which works independent of
```
Comment (ianvernon) [inline Documentation/gettingstarted/k8s-installers.rst]: independently

File: Documentation/kubernetes/requirements.rst [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+     bpffs			/sys/fs/bpf		bpf	defaults 0 0
+
+If you are using systemd to manage the kubelet, see the section
+:ref:`bpffs_systemd`.
+
+Enable automatic node CIDR allocation (Recommended)
+===================================================
+
+Kubernetes has the capability to automatically allocate and assign per node IP
```
Comment (ianvernon) [inline Documentation/kubernetes/requirements.rst]: assign a per-node

**Review by ianvernon — COMMENTED** [review]
Minor nits. Rest LGTM

File: Documentation/kubernetes/requirements.rst (hunk 2 above)
Comment (joestringer) [reply Documentation/kubernetes/requirements.rst]: What is the point of this notes column? It seems to have no notes at all..?

Comment (tgraf) [reply Documentation/kubernetes/requirements.rst]: I figured I would have notes to add as I complete the other sections but that never happend ;-). I've converted the table into a list.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-docs-please

**Pull request merged**